include/aspect/parameters.h

```diff
@@ -373,6 +373,50 @@ namespace aspect
       }
     };
 
+    /**
+     * A data structure containing a description of each compositional field.
+     * At present, this structure only includes the field type
+     * (i.e., whether it is of type chemical_composition, porosity, etc.).
+     */
+    struct CompositionalFieldDescription
+    {
+      /**
+       * This enum lists available compositional field types.
+       */
+      enum Type
+      {
+        chemical_composition,
+        stress,
+        grain_size,
+        porosity,
+        generic
+      } type;
+
+      /**
+       * This function translates an input string into the
+       * available enum options for the type of compositional field.
+       */
+      static
+      Type
+      parse_type(const std::string &input)
+      {
+        if (input == "chemical_composition")
```

Comment:
I would prefer the input file to be free of underscores and use spaces instead. Internally `chemical_composition` is the right name, but in the input file `chemical composition` is more consistent with all our other parameters. Same goes for grain_size below. Sorry I didnt see that in my earlier review.